File: mockez-back/src/main/java/com/mockez/domain/model/entity/GroupMember.java
package com.mockez.domain.model.entity;

import com.mockez.domain.model.entity.composite_key.GroupMemberPK;
import com.mockez.domain.model.entity.enumeration.Role;
import lombok.*;
import lombok.experimental.SuperBuilder;

import javax.persistence.Table;
import javax.persistence.*;

/**
 * @author Luong Quoc Trung, Do Quoc Viet
 */

@Getter
@Setter
@Entity
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "[GROUP_MEMBER]")
@ToString(callSuper = true)
@SuperBuilder(toBuilder = true)
public class GroupMember extends Base {

    @EmbeddedId
    private GroupMemberPK id;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private Role role = Role.GROUP_ASSOCIATE;

    @ManyToOne
    @ToString.Exclude
    @JoinColumn(nullable = false, foreignKey = @ForeignKey(name = "USER_GROUP_MEMBER_FK"), insertable = false, updatable = false)
    private User user = User.builder().build();

    @ManyToOne
    @ToString.Exclude
    @JoinColumn(nullable = false, foreignKey = @ForeignKey(name = "GROUP_GROUP_MEMBER_FK"), insertable = false, updatable = false)
    private Group group = Group.builder().build();

}

File: mocker-front/src/app/core/model/group-member.ts
import { Base } from '@core/model/base';
import { Role } from '@core/model/enumeration/role';
import { User } from '@core/model/user';
import { Group } from '@core/model/group';
import { GroupMemberPk } from '@core/model/composite_key/group-member-pk';

/**
 * @author Do Quoc Viet
 */

export class GroupMember extends Base {
  public id: GroupMemberPk;
  public role: Role;
  public user: User;
  public group: Group;
}
